### ISSUE TITLE
[Improve] Consider adding types for the package

### ISSUE THREAD

**Issue by edumserrano** [cenfun/monocart-reporter#43]
Perhaps the title of this issue is not the best, I'm not the most experienced front-end developer. Let me explain what I'm seeing:

1) I have a `TypeScript` project with `eslint`. The `eslint` configuration has lots of rules enabled.
2) I want to use the  `attachAuditReport` function so I do:
```ts
// near the top of my file, near the imports I have
const { attachAuditReport } = require("monocart-reporter");

// and in some tests I'll have
await attachAuditReport(<some params>);
```

When I do the above my `eslint` configuration warns me about the following rule:
> [@typescript-eslint/no-require-imports](https://typescript-eslint.io/rules/no-require-imports/):  Disallow invocation of require(). Prefer the newer ES6-style imports over require().

For now what I've done was change the usage of `require` to `import` such as `import { attachAuditReport } from "monocart-reporter";` and then I created a `monocart-reporter.d.ts` file which only contains `declare module "monocart-reporter";`. 

This does get rid of the warning but it's only a workaround. 

Suggestion: I believe you could consider updating your package to support ESM format? Or is this the wrong suggestion and perhaps there is another way to solve this issue I'm getting? 

If it helps, I found this link [Migrating an NPM package to use ES Modules](https://medium.com/digio-australia/migrating-an-npm-package-to-use-es-modules-d59877963d61).

**Comment by cenfun** [comment]
The package should support both `ESM` and `CommonJS`, but there is no type declaration for `Typescript`

**Comment by edumserrano** [comment]
Have you tried to use an `import` instead of a `require` in a TypeScript project ? Perhaps I'm doing something wrong but when I change to use `import` I get this error:

![image](https://github.com/cenfun/monocart-reporter/assets/15857357/7ab5b8bf-1da4-4a29-a000-4308b713c7d2)

I solved the error with creating this file:

monocart-reporter.d.ts
```ts
declare module "monocart-reporter" {
    import { TestInfo } from "@playwright/test";
    import { RunnerResult } from "lighthouse";

    export function attachAuditReport(
        lightHouseRunnerResult: RunnerResult,
        testInfo: TestInfo,
    ): Promise<void>;
}
```

So my suggestion about supporting ESM seems to be incorrect if you already support it. But if that's true then I don't know why I get the above error? Am I doing something wrong? Even if you don't have any type declarations it should be able to do the import without the above error right?

**Comment by edumserrano** [comment]
I think also the error I get is because I have a stricter eslint/tsconfig setup. I understand what you mean now, the package does support ESM or else it wouldn't even work on my project I guess. The warning is like you said because you don't provide any type declarations for TypeScript.

Feel free to close this issue unless you want to consider providing the types for TypeScript 😋 but I imagine that would be a mini project on its own...

**Comment by cenfun** [comment]
yes I consider adding the type declarations to package

**Comment by cenfun** [comment]
The types should be supported now, try version `1.7.0`
https://github.com/cenfun/monocart-reporter/blob/main/lib/index.d.ts

**Comment by edumserrano** [comment]
Tested and can confirm this is resolved on `1.7.0`. Thank you @cenfun 👏
